### ISSUE TITLE
VideoPress: module is still activated when downgrading plan

### ISSUE THREAD

**Issue by mmtr** [Automattic/jetpack#11233]
#### Steps to reproduce the issue

0. Start from a Jetpack site with an active Premium upgrade.
1. Go to Jetpack > Settings, and enable the VideoPress feature.
2. Remove the Premium upgrade from your account.
3. Go to Media > Library.
4. Upload a new video.

#### What I expected

Video should be uploaded correctly and not processed by VideoPress.

#### What happened instead

VideoPress intercepted the upload, but the video is not uploaded, probably because the API detects that we're missing an upgrade.

<img width="505" alt="screen shot 2019-01-31 at 11 56 51" src="https://user-images.githubusercontent.com/1233880/52050292-63979280-2550-11e9-99f6-49e22d3e4816.png">

#### Additional info

I found this when testing https://github.com/Automattic/jetpack/pull/11193. 

I noted that Jetpack loads the `/modules/videopress.php` file when we downgrade from Premium to Free. Furthermore, this stops happening as soon as you visit "Jetpack > Settings", so it seems that the VideoPress module is not deactivated during the downgrade, and we deactivate it when visiting the "Jetpack > Settings" page.

**Comment by jeherve** [comment]
Good catch. Maybe we should shortcircuit VideoPress features a bit earlier than here, to make sure that even if the plan and module status don't match, we don't generate any errors:
https://github.com/Automattic/jetpack/blob/b4437f5237e28390586b3d682dd80d97b68ccfef/modules/videopress/class.jetpack-videopress.php#L45
